crates/rust-analyzer/src/config.rs

```diff
@@ -89,6 +89,11 @@ config_data! {
         /// Internal config for debugging, disables loading of sysroot crates.
         cargo_noSysroot: bool            = "false",
 
+        /// Advanced option, fully override the command rust-analyzer uses for
+        /// checking buildscripts and procedural macros. The command should
+        /// include `--message-format=json` or similar option.
+        checkBuildScriptOnSave_overrideCommand: Option<Vec<String>> = "null",
```

Comment:
I suggest we call this `rust-analyzer.cargo.runBuildScriptsCommand` (maybe `runBuildScripts.command` would be ideal if we didn't already have `runBuildScripts` as a boolean option). For the description how about something like this:
```
Advanced option, fully override the command rust-analyzer uses to run build scripts and build procedural macros. The command should include `--message-format=json` or a similar option.
```

Reply:
done.